kdTree.js

```diff
@@ -9,52 +9,184 @@
  * @license MIT License <http://www.opensource.org/licenses/mit-license.php>
  */
 
- (function (root, factory) {
-    if (typeof define === 'function' && define.amd) {
-        define(['exports'], factory);
-    } else if (typeof exports === 'object') {
-        factory(exports);
-    } else {
-        factory(root);
-    }
+/*
+function assert(condition, message) {
+	if (condition) {
+		return;
+	}
+	message = message || "Assertion failed";
+	if (typeof Error !== "undefined") {
+		throw new Error(message);
+	}
+	throw message; // Fallback
+}*/
+
+function swap(points, i,j) {
+	const t = points[i];
+	points[i] = points[j];
+	points[j] = t;
+}
+
+// left is always included, right is always excluded,
+// so the range [left,right) contains (right-left) elements.
+function partition(points, left, right, pivIdx, dim) {
+	// Partition around the value found at position pivIdx.
+	const pivot = points[pivIdx][dim]; // get pivot value
+	var storeIdx = left; // variable to store the final position of the pivot value.
+	swap(points, pivIdx, right-1); // Move pivot to end
+	for (i=left; i < right-1; ++i) { // check all values but the last
```

Comment:
```suggestion
	for (let i=left; i < right-1; ++i) { // check all values but the last
```
Pretty sure `i` should be defined here.

Reply:
In fact, all of the for loops I've seen have undefined variables.

Reply:
Thank you very much @TheColorman  Is it standard practice nowadays in Javascript? (the code above is the only code I have ever written in JS)

Reply:
Yeah, the ability to use variables without declaration is considered a mistake in the language that exists due to backwards compatibility afaik.

Reply:
OK, I added `let` in every for loop. Anyway... after all these years, I doubt Ubilabs has any interest in my patch :-)